docs/clients/promtail/stages/replace.md

```diff
@@ -0,0 +1,184 @@
+# `replace` stage
+
+The `replace` stage is a parsing stage that parses a log line using a regular
+expression and replaces the log line. Named capture groups in the regex support adding data into the
+extracted map.
+
+## Schema
+
+```yaml
+replace:
+  # The RE2 regular expression. Each named capture group will be added to extracted.
+  # Each capture group and named capture group will be replaced with the value given in `replace`
+  expression: <string>
+
+  # Name from extracted data to parse. If empty, uses the log message.
+  # The replaced value will be assigned back to soure key
+  [source: <string>]
+
+  # Value to which the captured group will be replaced. The captured group or the named captured group will be
```

Comment:
what do you mean by or ? you mean all captured (including named ones) will be replaced ?

Reply:
Yeah, I meant both captured and named captured will be replaced. Will modify my statement